src/main/java/org/apache/commons/configuration2/AbstractConfiguration.java

```diff
@@ -429,6 +443,22 @@ public final boolean containsKey(final String key) {
      */
     protected abstract boolean containsKeyInternal(String key);
 
+    /**
+     * Returns true if this configuration contains one or more matches to this value. This operation stops at first
+     * match but may be more expensive than the {@link #containsKeyInternal containsKey} method.
+     * <p>
+     * The implementation of this method will be different depending on the type of Configuration used.
+     *
+     * <p>Note that this method is identical in functionality to
+     * {@link #containsValue containsValue}, (which is part of the {@link ImmutableConfiguration} interface).
+     *
+     * @param value a value to search for
+     * @return {@code true} if and only if some key maps to the {@code value} argument in this hashtable as determined
+     * by the {@code equals} method; {@code false} otherwise.
+     * @throws NullPointerException if the value is {@code null}
+     */
```

Comment:
New elements that are public or private should be documented with a Javadoc `@since` tag.

Reply:
How about protected ones? (sorry I know you didn't mention but just want to make sure)

Reply:
Thank you I wasn't aware of this rule until this PR, thank you again for clarifying will do that from now on.